tests/cloner_test.go

```diff
@@ -75,6 +76,166 @@ var _ = Describe("[rfe_id:1277][crit:high][vendor:cnv-qe@redhat.com][level:compo
+		Expect(err).To(HaveOccurred())
+		Expect(strings.Contains(err.Error(), "Target resources requests storage size is smaller than the source")).To(BeTrue())
+
+		By("Cloning from the source DataVolume to properly sized target")
+		targetDv = utils.NewDataVolumeForImageCloning("target-dv", "500Mi", f.Namespace.Name, sourceDv.Name)
+		_, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDv)
+		Expect(err).ToNot(HaveOccurred())
+		By("Waiting for clone to be completed")
+		err = utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, targetDv.Name)
+		Expect(err).ToNot(HaveOccurred())
+		matchFile := filepath.Join(testBaseDir, "disk.img")
+		Expect(f.VerifyTargetPVCContentMD5(f.Namespace, utils.PersistentVolumeClaimFromDataVolume(targetDv), matchFile, md5sum[:32])).To(BeTrue())
+	})
+})
+
+var _ = Describe("Validate Data Volume should clone multiple clones in parallel", func() {
+	f := framework.NewFrameworkOrDie(namespacePrefix)
+	tinyCoreIsoURL := fmt.Sprintf(utils.TinyCoreIsoURL, f.CdiInstallNs)
+
+	var (
+		sourceDv, targetDv1, targetDv2, targetDv3 *cdiv1.DataVolume
+		err                                       error
+	)
+
+	AfterEach(func() {
+		dvs := []*cdiv1.DataVolume{sourceDv, targetDv1, targetDv2, targetDv3}
+		for _, dv := range dvs {
+			cleanDv(f, dv)
+		}
+	})
+
+	It("[rfe_id:CNV-1277][case_id:CNV-1899][crit:High][vendor:cnv-qe@redhat.com][level:component] Should allow multiple cloning operations in parallel", func() {
+		By("Creating a source from a real image")
+		sourceDv = utils.NewDataVolumeWithHTTPImport("source-dv", "500Mi", tinyCoreIsoURL)
+		_, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, sourceDv)
+		Expect(err).ToNot(HaveOccurred())
+		By("Waiting for import to be completed")
+		utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, sourceDv.Name)
+
+		By("Calculating the md5sum of the source data volume")
+		md5sum, err := f.RunCommandAndCaptureOutput(utils.PersistentVolumeClaimFromDataVolume(sourceDv), "md5sum /pvc/disk.img")
+		Expect(err).ToNot(HaveOccurred())
+		fmt.Fprintf(GinkgoWriter, "INFO: MD5SUM for source is: %s\n", md5sum[:32])
+
+		// By not waiting for completion, we will start 3 transfers in parallell
+		By("Cloning from the source DataVolume to target1")
+		targetDv1 = utils.NewDataVolumeForImageCloning("target-dv1", "500Mi", f.Namespace.Name, sourceDv.Name)
+		_, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDv1)
+		Expect(err).ToNot(HaveOccurred())
+		By("Cloning from the source DataVolume to target2 in parallel")
+		targetDv2 = utils.NewDataVolumeForImageCloning("target-dv2", "500Mi", f.Namespace.Name, sourceDv.Name)
+		_, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDv2)
+		Expect(err).ToNot(HaveOccurred())
+		By("Cloning from the target1 DataVolume to target3 in parallel")
```

Comment:
Is it possible that this clone would start before the first one (source -> target1)?  In that case you'd have an empty target1 volume being cloned to target3

Reply:
Yes, that is a good source of heisenbugs. Let me change that to just be the source.